### DISCUSSION TITLE
Global Package Mapping in Virtual Environments

### DISCUSSION THREAD

**Discussion by aswnss-m** [pypa/virtualenv#2573] [Ideas]
### Introduction
This discussion focuses on the proposal to introduce a new feature, global package mapping, within project-specific virtual environments. The goal is to optimize disk space usage and enhance package management capabilities for developers working with virtual environments.

### Description
The proposed feature aims to automatically map globally installed packages to project-specific virtual environments. This mapping would allow developers to utilize globally installed packages within their virtual environments, reducing duplication of packages and optimizing disk space usage.

The key functionalities of this feature include:
- Automatic Mapping: When a project-specific virtual environment is activated, globally installed packages are automatically mapped to the virtual environment's `sys.path`.
- Package Availability Check: The virtual environment checks if a required package is already available globally and maps it accordingly, without requiring separate installation.
- Optional Installation: In cases where a required package is not globally available, developers would have the option to install it either within the virtual environment or globally, based on their preference.
- Seamless Dependency Management: The feature should seamlessly integrate with existing dependency management tools, allowing developers to specify project-specific dependencies using common formats like requirements files.

### Benefits
The introduction of global package mapping in virtual environments offers several benefits, including:
- Reduced Disk Space: Developers can utilize globally installed packages within multiple project-specific virtual environments, minimizing duplication of packages and optimizing disk space usage.
- Streamlined Workflow: The feature simplifies package management, reduces installation time, and ensures consistent access to global packages across projects.
- Improved Compatibility: Global package mapping enhances compatibility between projects that rely on the same global packages but require different virtual environments due to other project-specific dependencies.

**Comment by gaborbernat** [comment]
We do not plan too add this part of the core. You're encourage to create first plugin that achieve the same and if it gets traction we can reopen this topic.
